atom/browser/api/atom_api_app.cc

```diff
@@ -177,6 +197,37 @@ void App::OnFinishLaunching() {
   Emit("ready");
 }
 
+void App::OnSelectCertificate(
+    content::WebContents* web_contents,
+    net::SSLCertRequestInfo* cert_request_info,
+    scoped_ptr<content::ClientCertificateDelegate> delegate) {
+  v8::Isolate* isolate = v8::Isolate::GetCurrent();
+  v8::Locker locker(isolate);
+  v8::HandleScope handle_scope(isolate);
+  std::vector<mate::Dictionary> cert_info_list;
+
+  for (auto& cert : cert_request_info->client_certs) {
+    std::string encoded_data;
+    if (cert->valid_expiry().is_null() || cert->HasExpired()
```

Comment:
We should pass the expired certs to users since they might want to use an expired cert.